src/lib/progress-bar/progress-bar.ts

```diff
@@ -113,6 +131,21 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  /** Attaches value bar element animation end listener. */
+  ngAfterViewInit() {
+    if(this.mode === 'determinate' || this.mode === 'buffer') {
+      this._valueAnimationSubscription = fromEvent(this.primaryValueBar.nativeElement, 'transitionend')
+                                          .subscribe(() => this.valueAnimationEnd.next(this.value));
+    }
+  }
+
+  /** Removes value bar element animation end listener. */
```

Comment:
This comment can be omitted; it's pretty clear/obvious what `ngOnDestroy` is doing

Reply:
done.